src/Core/Util/SymfonyArrayFinder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+class SymfonyArrayFinder implements \ArrayAccess, \Countable
```

Comment:
Note: I'm not sure I can use `declare(strict_types=1)` as I need to match the old class public API.

Reply:
Why `SymfonyArrayFinder` and not just `ArrayFinder`?

Reply:
> Note: I'm not sure I can use `declare(strict_types=1)` as I need to match the old class public API.

I don't understand why would that be a problem

Reply:
I don't understand why `Symfony` in the class name :thinking: 
And same as @eternoendless I also don't understand how it could be a problem :thinking:

Reply:
> Maybe not, but in any case hidden errors will surface, and that's precisely the reason we choose to enforce strict types.

OK then, my ArrayFinder will be a little bit different from its predecessor. Cleaner but riskier. But if you want it this way 😄  I'm in!

Will remove the Symfony from name too.

Reply:
Done in https://github.com/PrestaShop/PrestaShop/pull/25611/commits/c0066eb57f7c1f3ed17205f820cc4a0a789c5073